rest-java/src/main/java/org/hiero/mirror/restjava/mapper/FeeScheduleMapper.java

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package org.hiero.mirror.restjava.mapper;
+
+import com.hederahashgraph.api.proto.java.CurrentAndNextFeeSchedule;
+import com.hederahashgraph.api.proto.java.ExchangeRateSet;
+import com.hederahashgraph.api.proto.java.HederaFunctionality;
+import com.hederahashgraph.api.proto.java.TransactionFeeSchedule;
+import java.util.Comparator;
+import java.util.Map;
+import org.hiero.mirror.rest.model.NetworkFee;
+import org.hiero.mirror.rest.model.NetworkFeesResponse;
+import org.hiero.mirror.restjava.dto.SystemFile;
+import org.jspecify.annotations.Nullable;
+import org.mapstruct.Mapper;
+
+@Mapper(config = MapperConfiguration.class, uses = CommonMapper.class)
+public interface FeeScheduleMapper {
+
+    long FEE_DIVISOR_FACTOR = 1000L;
+
+    Map<HederaFunctionality, String> ENABLED_TRANSACTION_TYPES = Map.of(
+            HederaFunctionality.ContractCall, "ContractCall",
+            HederaFunctionality.ContractCreate, "ContractCreate",
+            HederaFunctionality.EthereumTransaction, "EthereumTransaction");
+
+    default NetworkFeesResponse map(
+            SystemFile<CurrentAndNextFeeSchedule> feeScheduleFile,
+            SystemFile<ExchangeRateSet> exchangeRateFile,
+            String order,
+            CommonMapper commonMapper) {
+
+        var schedule = feeScheduleFile.protobuf().getCurrentFeeSchedule();
+        var rate = exchangeRateFile.protobuf().getCurrentRate();
+
+        var fees = schedule.getTransactionFeeScheduleList().stream()
+                .filter(s -> ENABLED_TRANSACTION_TYPES.containsKey(s.getHederaFunctionality()) && s.getFeesCount() > 0)
+                .map(s -> mapToNetworkFee(s, rate.getHbarEquiv(), rate.getCentEquiv()))
```

Comment:
Can pass a single rate param through to mapToNetworkFee and convertGasPriceToTinyBars

```suggestion
                .map(s -> mapToNetworkFee(s, rate))
```